cmake/Dependencies.cmake

```diff
@@ -1088,7 +1085,12 @@ if(USE_ROCM)
   else()
     caffe2_update_option(USE_ROCM OFF)
   endif()
+endif()
 
+# ---[ ROCm
+if(USE_ROCM)
+  # We check again for USE_ROCM because it might have been set to OFF
+  # in the if above
```

Comment:
Before this change, the first run of `cmake` will add `-isystem /opt/rocm/hip/include (...)` to the compilation command whether ROCM is present or not. It will be removed on the second and subsequent runs, which causes a lot of unnecessary rebuilding.

Reply:
Eeeew. Can we just make it a hard error to enable HIP instead of toggling it OFF above?

Reply:
The CI depends on USE_ROCM being gracefully disabled. When I tried to hard error, there were 16 CI failures.

I could disable USE_ROCM by default and require the user to override it if desired. Would you like me to do that?

Reply:
Let's take this as is for now. I see CUDA is doing a similar thing.